tools/svg2c/main.go

```diff
@@ -0,0 +1,298 @@
+	}
+	return parsedPoints
+}
+
+func (sd *SvgData) CalcOrigin() {
+	minPoint := sd.GetMinPoint()
+	maxPoint := sd.GetMaxPoint()
+	midPoint := Point{
+		(maxPoint.X - minPoint.X) / 2,
+		(maxPoint.Y - minPoint.Y) / 2,
+	}
+	sd.Origin = Add(minPoint, midPoint)
+}
+
+func (sd *SvgData) CalcDataWithOffset() {
+	for idx, gData := range sd.Data {
+		sd.Data[idx].DeltaPoints = gData.GetWithOffsetAndDelta(sd.Origin)
+	}
+}
+
+func (sd *SvgData) GetMinPoint() Point {
+	var minPoint = sd.Data[0].Points[0]
+	for _, gData := range sd.Data {
+		for _, point := range gData.Points {
+			if point.X < minPoint.X {
+				minPoint.X = point.X
+			}
+			if point.Y < minPoint.Y {
+				minPoint.Y = point.Y
+			}
+		}
+	}
+	return minPoint
+}
+
+func (sd *SvgData) GetMaxPoint() Point {
+	var maxPoint = sd.Data[0].Points[0]
+	for _, gData := range sd.Data {
+		for _, point := range gData.Points {
+			if point.X > maxPoint.X {
+				maxPoint.X = point.X
+			}
+			if point.Y > maxPoint.Y {
+				maxPoint.Y = point.Y
+			}
+		}
+	}
+	return maxPoint
+}
+
+func (sd *SvgData) Export() (output string, err error) {
+	t, err := template.New("svg").Parse(svgTemplate)
+	if err != nil {
+		return
+	}
+
+	file, err := os.Create(outputPath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	var data bytes.Buffer
+	err = t.Execute(&data, sd)
+	if err != nil {
+		return "", err
+	}
+	return data.String(), nil
+}
+
+func (gd *GeometricData) GetWithOffsetAndDelta(offset Point) []Point {
+	inputPoints := gd.GetPointsWithOffset(offset)
+	deltaPoints := []Point{inputPoints[0]}
+	for index, point := range inputPoints {
+		if index != 0 {
+			lastPoint := inputPoints[index-1]
+			deltaPoints = append(deltaPoints, Sub(point, lastPoint))
+		}
+	}
+	return deltaPoints
+}
+
+func (gd *GeometricData) GetPointsWithOffset(offset Point) []Point {
+	var withOffset []Point
+	for _, point := range gd.Points {
+		withOffset = append(withOffset, Sub(point, offset))
+	}
+	return withOffset
+}
+
+func handleFile(file *os.File, name string) string {
+	svgData := SvgData{[]GeometricData{}, Point{0, 0}, name}
+
+	element, _ := svgparser.Parse(file, false)
+	elements := element.Children
+
+	for _, element := range elements {
+		elementType := element.Name
+		switch elementType {
+		case "polyline":
+			parsedPoints := parsePolyLine(element)
+			svgData.Data = append(svgData.Data,
+				GeometricData{len(parsedPoints), parsedPoints, []Point{}})
+		case "line":
+			parsedPoints := parseLine(element)
+			svgData.Data = append(svgData.Data,
+				GeometricData{len(parsedPoints), parsedPoints, []Point{}})
+		case "path":
+			if verbose {
+				fmt.Println("Found path element, geometry won't be parsed")
+			}
+		default:
+			if verbose {
+				fmt.Printf("[WARN] Parsed element %s\n", element.Name)
+			}
+		}
+	}
+	svgData.CalcOrigin()
+	svgData.CalcDataWithOffset()
+	data, err := svgData.Export()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return data
+}
+
+const defaultFileName = "data.c"
```

Comment:
IMO everything above this line should be in a separate file. And if you don't want to do that, then please at least change function's names to lowercase (`CalcDataWithOffset()` or `GetPointsWithOffset()` for example) so they won't be needlessly exported